diff --git a/debian/changelog b/debian/changelog
--- a/debian/changelog
+++ b/debian/changelog
@@ -1 +1,13 @@
+gnutls28 (3.8.10-2) unstable; urgency=medium
+
+  * Change the generate of the FIPS HMAC to be after the dh_strip phase.
+
+ -- Jan Alexander Preissler <jan.preissler@sap.com>  Tue Oct 28 15:38:23 2025 +0100
+
+gnutls28 (3.8.10-2) unstable; urgency=medium
+
+  * Enable FIPS configuration option.
+
+ -- Jan Alexander Preissler <jan.preissler@sap.com>  Fri Oct 24 15:55:31 2025 +0200
+
 gnutls28 (3.8.10-2) unstable; urgency=medium
