--- a/debian/rules	2026-02-12 18:30:12.738690170 +0100
+++ b/debian/rules	2026-02-12 18:31:38.274215991 +0100
@@ -7,9 +7,7 @@ export DEB_CXXFLAGS_MAINT_APPEND := -Wal
 
 include /usr/share/dpkg/pkg-info.mk
 
-AMCONFBUILDINDEP := $(shell if dh_listpackages | grep -q gnutls-doc ; \
-	then echo "--enable-gtk-doc" ; \
-	else echo "--disable-gtk-doc --disable-doc"; fi)
+AMCONFBUILDINDEP := --disable-gtk-doc --disable-doc
 
 CONFIGUREARGS = \
 	--enable-ld-version-script --enable-cxx \
@@ -33,13 +31,7 @@ override_dh_auto_configure:
 
 override_dh_autoreconf:
 	grep -rl --null gettext-0.20 m4/ | xargs -0r rm -v
-	if ! dh_listpackages | grep -q gnutls-doc ; \
-		then env GTKDOCIZE="echo DISABLED running gtkdocize" \
-			dh_autoreconf --verbose $(BDIR) ; \
-		else \
-			dh_autoreconf --verbose $(BDIR) ; \
-		fi
-	# Bug#1118328
+	dh_autoreconf --verbose $(BDIR)
 	cp -v debian/texinfo.tex build-aux/
 
 override_dh_makeshlibs:
@@ -84,35 +76,17 @@ override_dh_auto_build:
 	touch --reference=lib/includes/gnutls/abstract.h \
 		lib/includes/gnutls/gnutls.h.in
 	dh_auto_build $(BDIR) --verbose
-ifeq ($(filter --disable-doc,$(AMCONFBUILDINDEP)),)
-	$(MAKE) -C b4deb html
-	rm -f doc/gnutls.pdf && $(MAKE) -C b4deb/doc gnutls.pdf
-else
 	$(MAKE) -C b4deb/doc/manpages
-endif
 
 override_dh_auto_install:
 	dh_auto_install $(BDIR) --verbose
-ifneq ($(filter --disable-doc,$(AMCONFBUILDINDEP)),)
 	$(MAKE) -C b4deb/doc/manpages DESTDIR=$(CURDIR)/debian/tmp install
-else
-	$(MAKE) -C b4deb/doc/ DESTDIR=$(CURDIR)/debian/tmp install-html
-	# we symlink these
-	rm -vf debian/tmp/usr/share/info/*.png
-endif
 	find debian/*/usr/lib/* -name '*.so.*.*' -type f -exec \
 		chrpath -d {} +
 	find debian/*/usr/lib/ -name 'libgnutlsxx.*' -delete
 	# We need to prepeare this, since dh_install expects to copy a file. 
 	touch debian/tmp/usr/lib/$(shell uname -m)-linux-gnu/.libgnutls.so.30.hmac
 
-override_dh_installinfo:
-	dh_installinfo $(BDIR)
-	if test -e debian/gnutls-doc ; then \
-		cd debian/gnutls-doc/usr/share/info && \
-		sed -i -e 's:image src="\([^"]*.png"\):image src="/usr/share/doc/gnutls-doc/html/\1:g' *.info* ; \
-	fi
-
 override_dh_compress:
 	dh_compress $(BDIR) -X.pdf
 
--- a/debian/gnutls-doc.install	2026-02-12 20:54:36.533650203 +0100
+++ b/debian/gnutls-doc.install	2026-02-12 20:54:46.144641496 +0100
@@ -1,7 +1,2 @@
-usr/share/doc/gnutls/*.html usr/share/doc/gnutls-doc/html
-usr/share/doc/gnutls/*.png usr/share/doc/gnutls-doc/html
-usr/share/gtk-doc/html/gnutls/*.css usr/share/doc/gnutls-doc/api-reference
-usr/share/gtk-doc/html/gnutls/*.devhelp* usr/share/doc/gnutls-doc/api-reference
-usr/share/gtk-doc/html/gnutls/*html usr/share/doc/gnutls-doc/api-reference
-usr/share/gtk-doc/html/gnutls/*png usr/share/doc/gnutls-doc/api-reference
+
 
