diff --git a/debian/rules b/debian/rules
--- a/debian/rules
+++ b/debian/rules
@@ -14,2 +14,5 @@ AMCONFBUILDINDEP := $(shell if dh_listpackages | grep -q gnutls-doc ; \
 CONFIGUREARGS = \
+	--enable-fips140-mode \
+	--with-fips140-module-name='Garaden Linux gnutls $(DEB_VERSION)' \
+	--with-fips140-module-version=$(DEB_VERSION) \
 	--enable-ld-version-script --enable-cxx \
@@ -82,9 +85,5 @@ endif
 	find debian/*/usr/lib/ -name 'libgnutlsxx.*' -delete
-
-override_dh_installinfo:
-	dh_installinfo $(BDIR)
-	if test -e debian/gnutls-doc ; then \
-		cd debian/gnutls-doc/usr/share/info && \
-		sed -i -e 's:image src="\([^"]*.png"\):image src="/usr/share/doc/gnutls-doc/html/\1:g' *.info* ; \
-	fi
+	# Generate the HMAC files that we need for the selftest.
+	b4deb/lib/fipshmac > debian/tmp/usr/lib/$(shell uname -m)-linux-gnu/.libgnutls.so.30.hmac
+	sed -i 's|/tmp/tmp[^/]*/src/b4deb/lib/.libs/|/usr/lib/$(shell uname -m)-linux-gnu/|g' debian/tmp/usr/lib/$(shell uname -m)-linux-gnu/.libgnutls.so.30.hmac
